content/doc/developer/plugin-development/incrementals.adoc

```diff
@@ -0,0 +1,108 @@
+---
+layout: developersection
+title: "Developing plugins and Jenkins core in parallel"
+tags:
+- jenkins
+- incrementals
+- core
+- plugin
+description: "This article explains how to use Incrementals for developing plugins in parallel with Jenkins core"
+---
+
+A common problem one might stumble upon is developing an API in Jenkins core and simultaneously
+building a reference implementation inside a plugin.
+What to do, one might ask.
+
+Worry not, Incrementals are here to the rescue!
+
+With incrementals, we can create Jenkins core incremental versions which can be used along with the plugin to build the
+plugin against the very latest changes you have made to core.
+Also, it allows keeping track and switching between incremental versions.
+Not only that, it allows others to also not have to build Jenkins locally to try out the changes or collaborate with
+you on your awesome project!
+
+Sounds cool? Let's see how to go about it.
+
+== 1. Create a Pull Request in Jenkins core
+
+First step is to link:https://github.com/jenkinsci/jenkins/pulls[create a PR on github] by clicking the "New Pull
+Request" button and push the changes to Jenkins core.
+
+image::/images/developer/plugin-development/incrementals/create-pr.png[]
+
+Note: Make sure that you rebase your branch with the latest changes from master, otherwise incrementals version
+will NOT be created.
+
+== 2. Wait for all checks to pass
+
+Wait for 'continuous-integration/jenkins' to perform checks, this can take about two hours.
+
+== 3. Obtain the Incrementals version
+
+After the checks are completed, you should get a message like below:
+
+image::/images/developer/plugin-development/incrementals/checks.png[]
+
+Click on 'Details' adjacent to 'continuous-integration/jenkins/incrementals — Deployed to Incrementals.'
+
+Note: If you only see one check 'continuous-integration/jenkins/pr-merge — This commit looks good' then there is an
+error due to which incrementals version was not created. One of the most likely reasons is that latest changes from
+master have not been pulled into the branch. You can click on 'Details' adjacent to it to investigate this error from
+the logs.
+
+== 4. Copy the Version ID
+
+You should be redirected to a page like below:
+
+image::/images/developer/plugin-development/incrementals/incrementals.png[]
+
+Copy the version ID, which in this example would be `2.238-rc29961.5c3c5871cca6`.
+
+== 5. Update the plugin POM
+
+There are two changes you want to make to your plugin POM.
+
+First, update the Jenkins version to the version ID copied in the previous step:
+
+```
+...
+    <properties>
+        <jenkins.version>2.238-rc29961.5c3c5871cca6</jenkins.version>
```

Comment:
Again, this is only applicable to the case of Jenkins core being upstream. For a plugin upstream, you would just use that version in the `<version>` of your `<dependency>`.